cmd/cert-checker/main.go

```diff
@@ -229,48 +225,52 @@ func (c *certChecker) checkCert(cert core.Certificate, ignoredLints map[string]b
 		}
-		// Check we have the right expiration time
+		// Check that we have the correct expiration time.
 		if !parsedCert.NotAfter.Equal(cert.Expires) {
 			problems = append(problems, "Stored expiration doesn't match certificate NotAfter")
 		}
-		// Check basic constraints are set
+		// Check if basic constraints are set.
 		if !parsedCert.BasicConstraintsValid {
 			problems = append(problems, "Certificate doesn't have basic constraints set")
 		}
-		// Check the cert isn't able to sign other certificates
+		// Check that the cert isn't able to sign other certificates.
 		if parsedCert.IsCA {
 			problems = append(problems, "Certificate can sign other certificates")
 		}
-		// Check the cert has the correct validity period. The validity period is
-		// computed inclusive of the whole final second indicated by notAfter.
-		validityPeriod := parsedCert.NotAfter.Add(time.Second).Sub(parsedCert.NotBefore)
-		if _, ok := c.acceptableValidityPeriods[uint(validityPeriod.Seconds())]; !ok {
-			problems = append(problems, fmt.Sprintf("Certificate has unacceptable validity period"))
+		// Check that the cert has a valid validity period. The validity
+		// period is computed inclusive of the whole final second indicated by
+		// notAfter.
+		validityDuration := parsedCert.NotAfter.Add(time.Second).Sub(parsedCert.NotBefore)
+		_, ok := c.acceptableValidityDurations[validityDuration]
+		if !ok {
+			problems = append(problems, "Certificate has unacceptable validity period")
+			problems = append(problems, fmt.Sprintf("Certificate %q has unacceptable validity period of '%d' nanoseconds", parsedCert.DNSNames, validityDuration.Nanoseconds()))
 		}
-		// Check the stored issuance time isn't too far back/forward dated
+		// Check that the stored issuance time isn't too far back/forward dated.
 		if parsedCert.NotBefore.Before(cert.Issued.Add(-6*time.Hour)) || parsedCert.NotBefore.After(cert.Issued.Add(6*time.Hour)) {
 			problems = append(problems, "Stored issuance date is outside of 6 hour window of certificate NotBefore")
 		}
-		// Check CommonName is <= 64 characters
+		// Check if the CommonName is <= 64 characters.
 		if len(parsedCert.Subject.CommonName) > 64 {
 			problems = append(
 				problems,
 				fmt.Sprintf("Certificate has common name >64 characters long (%d)", len(parsedCert.Subject.CommonName)),
 			)
 		}
-		// Check that the PA is still willing to issue for each name in DNSNames + CommonName
+		// Check that the PA is still willing to issue for each name in DNSNames
+		// + CommonName.
 		for _, name := range append(parsedCert.DNSNames, parsedCert.Subject.CommonName) {
 			id := identifier.ACMEIdentifier{Type: identifier.DNS, Value: name}
-			// TODO(https://github.com/letsencrypt/boulder/issues/3371): Distinguish
-			// between certificates issued by v1 and v2 API.
-			if err = c.pa.WillingToIssueWildcards([]identifier.ACMEIdentifier{id}); err != nil {
+			// TODO(#3371): Distinguish between certificates issued by v1 and v2 API.
```

Comment:
I think we could probably just delete this whole comment and the ticket that goes with it since v1 is no longer available and being deleted soon.

Reply:
Omg, that's totally true 👍🏻